### PR TITLE
Change the signature of array getIndices

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

File: src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs [hunk 1]
```diff
@@ -1102,18 +1102,17 @@ data GetIndicesState contents start end st =
 -- /Pre-release/
 {-# INLINE getIndices #-}
 getIndices :: (MonadIO m, Storable a) =>
-    Unfold m (Array a) Int -> Unfold m (Array a) a
-getIndices (Unfold stepi injecti) = Unfold step inject
+    D.Stream m Int -> Unfold m (Array a) a
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs:1105]: This should be `SerialT m Int`.

File: src/Streamly/Internal/Data/Array/Foreign.hs [hunk 2]
```diff
@@ -459,8 +459,9 @@ getIndex arr i =
 --
 -- /Unimplemented/
 {-# INLINE getIndices #-}
-getIndices :: Unfold m (Array a) Int -> Unfold m (Array a) a
-getIndices = undefined
+getIndices :: (MonadIO m, Storable a)
+    => SerialT m Int -> Unfold m (MA.Array a) a
+getIndices (SerialT stream) = MA.getIndices $ D.fromStreamK stream
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Array/Foreign.hs:464]: This is immutable array module, this API should use the immutable array type, not `MA.Array`.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Comment by rnjtranjan** [comment]
Fixes : #1402

File: src/Streamly/Internal/Data/Array/Foreign.hs [hunk 3]
```diff
@@ -457,10 +462,34 @@ getIndex arr i =
...
+            D.Stop -> do
+                liftIO $ touch (arrContents arr)
+                return D.Stop
+
 {-# INLINE getIndices #-}
-getIndices :: Unfold m (Array a) Int -> Unfold m (Array a) a
-getIndices = undefined
+getIndices :: (MonadIO m, Storable a)
+    => SerialT m Int -> Unfold m (Array a) a
+getIndices (SerialT stream) = getIndices' $ D.fromStreamK stream
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Array/Foreign.hs:492]: Just use unsafeFreeze/unsafeThaw to convert mutable/immutable arrays, do not duplicate the code.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Comment by harendra-kumar** [comment]
It would be nice to have a test as well, we can copy the takeWhile test if it exists.

**Pull request merged**